src/settings/AudioSettings.tsx

```diff
@@ -98,18 +62,63 @@ export default function GeneralSettings(props: ISettingsPanelProps) {
     return true;
   };
 
-  // Remove any unknown devices from the list, and the state variables. That
-  // means if the user presses save, even without any changes, the unknown
-  // devices will be removed from config.
-  React.useEffect(() => {
-    const knownInputs = input.filter(isKnownDevice);
-    setInput(knownInputs);
-    onChange(new FakeChangeEvent('audioInputDevices', knownInputs));
+  /**
+   * Standardizes device names to an array of strings and filters by known devices.
+   *
+   * @param deviceNames the device names to standardize
+   * @returns the standardized device names
+   */
+  const standardizeDeviceNames = (deviceNames: string[] | string): string[] => {
+    let normalizedDeviceNames: string[];
+
+    if (typeof deviceNames === 'string') {
+      normalizedDeviceNames = deviceNames.split(',');
+    } else {
+      normalizedDeviceNames = deviceNames;
+    }
+
+    return normalizedDeviceNames.filter(isKnownDevice);
+  };
+
+  let initialInputs: string[];
+
+  if (config.audioInputDevices) {
+    initialInputs = standardizeDeviceNames(config.audioInputDevices);
+  } else {
+    initialInputs = [];
+  }
+
+  let initialOutputs: string[];
+
+  if (config.audioOutputDevices) {
+    initialOutputs = standardizeDeviceNames(config.audioOutputDevices);
+  } else {
+    initialOutputs = [];
+  }
 
-    const knownOutputs = output.filter(isKnownDevice);
-    setOutput(knownOutputs);
-    onChange(new FakeChangeEvent('audioOutputDevices', knownOutputs));
-  }, []);
+  const forceMonoConfigKey: ConfigurationSchemaKey = 'obsForceMono';
+  const initialForceMono = config.obsForceMono ?? false;
```

Comment:
Do we need ?? here? Config from the ConfigService should always default to something sensible I think, even if it's not present

Reply:
No, good catch. I think I added it before I'd set the type of the config prop. Looks like I can remove some other similar conditionals for initializing values from config as well.